Tests/MessagingInApp/MessagingInAppIntegrationTests.swift

```diff
@@ -47,6 +47,48 @@ class MessagingInAppIntegrationTest: IntegrationTest {
         XCTAssertFalse(didCallGlobalEventListener)
     }
 
+    func test_givenNavigateToDifferentScreenWhileMessageAnimatingIntoView_expectDoNotShowModalMessage() {
```

Comment:
`ViewAnimationRunnerStub` allows us to have control over when an animation finishes running. 

Because of this, I wrote new tests for scenarios that could happen in an app. We expect the SDK dismisses a modal if the rendering is done but the modal has not animated into view yet.